src/ripple/basics/base_uint.h

```diff
@@ -549,103 +550,77 @@ using uint160 = base_uint<160>;
 using uint256 = base_uint<256>;
 
 template <std::size_t Bits, class Tag>
-inline int
-compare(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
+[[nodiscard]] inline constexpr std::strong_ordering
+operator<=>(base_uint<Bits, Tag> const& lhs, base_uint<Bits, Tag> const& rhs)
 {
-    auto ret = std::mismatch(a.cbegin(), a.cend(), b.cbegin());
+    // This comparison might seem wrong on a casual inspection because it
+    // compares data internally stored as std::uint32_t byte-by-byte. But
+    // note that the underlying data is stored in big endian, even if the
+    // plaform is little endian. This makes the comparison correct.
+    //
+    // FIXME: use std::lexicographical_compare_three_way once support is
+    //        added to MacOS.
 
-    if (ret.first == a.cend())
-        return 0;
+    auto const ret = std::mismatch(lhs.cbegin(), lhs.cend(), rhs.cbegin());
+
+    // a == b
+    if (ret.first == lhs.cend())
+        return std::strong_ordering::equivalent;
 
     // a > b
     if (*ret.first > *ret.second)
-        return 1;
+        return std::strong_ordering::greater;
 
     // a < b
-    return -1;
-}
-
-template <std::size_t Bits, class Tag>
-inline bool
-operator<(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
-{
-    return compare(a, b) < 0;
-}
-
-template <std::size_t Bits, class Tag>
-inline bool
-operator<=(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
-{
-    return compare(a, b) <= 0;
-}
-
-template <std::size_t Bits, class Tag>
-inline bool
-operator>(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
-{
-    return compare(a, b) > 0;
-}
-
-template <std::size_t Bits, class Tag>
-inline bool
-operator>=(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
-{
-    return compare(a, b) >= 0;
-}
-
-template <std::size_t Bits, class Tag>
-inline bool
-operator==(base_uint<Bits, Tag> const& a, base_uint<Bits, Tag> const& b)
-{
-    return compare(a, b) == 0;
+    return std::strong_ordering::less;
```

Comment:
Thanks. Fixed.

Reply:
Wait what? You can’t reduce lines 571-575 down to a single return.

Am I not understanding the suggestion here?

Reply:
The code is now like below.

```
    auto const ret = std::mismatch(lhs.cbegin(), lhs.cend(), rhs.cbegin());

    // a == b
    if (ret.first == lhs.cend())
        return std::strong_ordering::equivalent;

    return (*ret.first > *ret.second) ? std::strong_ordering::greater
                                      : std::strong_ordering::less;
```

We can use a ternary operator because we know that `a != b`. If you want to revert this change, let me know.